Libraries/Alert/AlertMacOS.js

```diff
@@ -175,6 +176,11 @@ class AlertMacOS {
     modal?: ?boolean,
     critical?: ?boolean,
   ): void {
+    warnOnce(
+      'deprecated-AlertMacOS',
+      '"AlertMacOS" module has been deprecated in favor of "Alert" and will be removed in a future version of react-native-macos',
+    );
+
```

Comment:
I checked and nobody downstream at Microsoft and on Github, and afaik nobody uses this. Maybe we can just delete it?

Reply:
We already warn about the usage of AlertMacOS in [index.js](https://github.com/microsoft/react-native-macos/blob/1ae2e06db25ebf32513d307aa7503c4f831cab8a/index.js#L274-L283), so this extra warning is unnecessary for standard RN development.

That being said, I'm okay with deleting this altogether. The warning's been around for [over two years](24be28ebb23f4b987f06aa6e967a9bfdd3a969bb); that's more than enough time for RN developers to migrate over.

Reply:
Let's remove it. That would also help us clean some diffs around the test pages here. I went ahead and made a (draft) upstream fix to help with this effort: https://github.com/facebook/react-native/pull/35586

Reply:
Both this PR and my upstream PR are ready now. I updated my PR notes to explain how they relate.